### PR TITLE
[SPARK-44264][PYTHON][ML][FOLLOW-UP] Clean Up Deepspeed Code

### PR DESCRIPTION
**Pull request by mathewjacob1002**
Follow up on @HyukjinKwon comments on [this pr](https://github.com/apache/spark/pull/41770).
### Checklist
- [x] Add docs on Deepspeed Distributor to the pyspark.ml.rst
- [x] Make all variable names lowercase in test_deepspeed_distributor per PEP8
- [ ] Make docstrings adhere to numpy standard
- [ ] Add an example of how to use deepspeed distributor in the examples section(same as TorchDistributor basically)
- [ ]

### CODE REVIEW

**Comment by mathewjacob1002** [comment]
@HyukjinKwon do you think we should add the examples only after we get everything in DeepspeedTorchDistributor feature-complete?

**Review by HyukjinKwon — APPROVED** [review]
I guess the user-facing example and guide are working in progress separately? Given that LGTM. For example, we should have `Examples` section at `DeepspeedTorchDistributor`'s docstring.

**Comment by HyukjinKwon** [comment]
Ah just read

> do you think we should add the examples only after we get everything in DeepspeedTorchDistributor feature-complete?

Yes, that works

File: python/docs/source/reference/pyspark.ml.rst [hunk 1]
```diff
@@ -351,6 +351,7 @@ Distributor
     :toctree: api/
 
     TorchDistributor
+    DeepspeedTorchDistributor
```
Comment (HyukjinKwon) [inline python/docs/source/reference/pyspark.ml.rst:354]: Seems like it fails:

```
2023-07-12T23:31:59.4320534Z Warning, treated as error:
2023-07-12T23:31:59.4322346Z [autosummary] failed to import 'pyspark.ml.torch.distributor.DeepspeedTorchDistributor': no module named pyspark.ml.torch.distributor.DeepspeedTorchDistributor
```

Comment (mathewjacob1002) [reply python/docs/source/reference/pyspark.ml.rst:354]: oh it's torch.deepspeed.DeepspeedTorchDistributor

**Comment by mathewjacob1002** [comment]
@HyukjinKwon Everything should work now :D could we merge this?

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**